app/components/reusables/input-box.hbs

```diff
@@ -1,17 +1,33 @@
-<div data-test-input class='input-box'>
-  <label data-test-label class='label' for={{@name}}>{{@field}}</label>
+<div data-test-input={{@name}} class='input-box input-box--btn'>
+  <label
+    data-test-label={{@name}}
+    class='label'
+    for={{@name}}
+  >{{@field}}</label>
   {{#if @required}}
-    <span data-test-required class='required'>*</span>
+    <span data-test-required={{@name}} class='required'>*</span>
   {{/if}}
   <input
-    data-test-input-field
+    data-test-input-field={{@name}}
     name={{@name}}
-    class='input__field user-input input-regular'
+    class='input__field user-input input-regular {{@classDisableInput}}'
     type={{@type}}
     id={{@name}}
     placeholder={{@placeHolder}}
     required={{@required}}
     value={{@value}}
+    disabled={{@disabled}}
     {{on 'input' @onInput}}
   />
+  {{#if @hasButtonInsideInput}}
+    <Reusables::Button
+      @text='Generate Username'
+      @variant='dark'
+      @classGenerateUsername='generateUsername'
+      @onClick={{@onClickGetUsername}}
+      @disabled={{@isValid}}
+      @test='generateUsername'
+      @type='button'
+    />
+  {{/if}}
```

Comment:
this if the condition is not required... please remove this if the condition is or else give me some test case where this condition is valid through the example

Reply:
it means i want to show button and used here condition because I want button should be inside div
Yes it is required when I want button in particular input field and given condition because using reusable Input component
Test. please check this test in which button is also render with input- https://github.com/Real-Dev-Squad/website-www/pull/622/files#diff-c28ca67ea4423a3a5efdf182d7182e47224aa1887c296cfe6b209149ac068aa1R94